File: Data Structures & Algorithms/two-integer-sum/submission-5.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        let map = new Map();
        const length = nums.length;
        for(let i=0; i<length; i++){
            if(map.has(target-nums[i])){
                return [map.get(target-nums[i]),i];
            }else{
                map.set(nums[i],i);
            }
        }
    }
}

File: Data Structures & Algorithms/binary-search/submission-5.java
class Solution {
    public int search(int[] nums, int target) {
        int ans = -1;
        int left=0;
        int right = nums.length-1;
        while(left<=right){
            int mid = (right-left)/2+left;
            if(nums[mid]<target){
                left=mid+1;
            }else if(nums[mid]>target){
                right=mid-1;
            }else{
                return mid;
            }
        }
        return ans;
    }
}
